webdriver/result.py

```diff
@@ -0,0 +1,31 @@
+import json
+
+class Result():
+    def __init__(self, response):
+        self.status = response.status
+        self.body = response.read()
+
+        if self.body:
+            self.body = json.loads(self.body)
+
+        self._validate(response)
```

Comment:
I dislike constructors that do too much processing as it weakens the trust in that an object can be constructed without adverse consequences. Can we create a static method on this object, `from_response` (or equivalent) that reads, parses, and validates it? Then it can pass the valid data pieces into this ctor.